### ISSUE TITLE
Explore many-to-many matches for profiles

### ISSUE THREAD

**Issue by jmcmurry** [monarch-initiative/owlsim-v3#89]
It would be interesting to test out the profile matching when ALL pairwise phenotype matches (N Factorial matches above a threshold) are factored into the overall profile match score. 

Eg, in the situation below, what would it change to NOT throw out the match between ...epiphysis and toe, simply because the one between epiphysis and finger is better? Same with the delayed speech and intellectual disability features.

![fig3_ws_syndrome_double_triptych_2017-05-14d](https://cloud.githubusercontent.com/assets/3658863/26227582/5e8a0b18-3be8-11e7-8cec-3ec0f902c3c4.png)

Thoughts? cc: @julesjacobsen , @damiansm, @mellybelly, @pnrobinson

**Comment by pnrobinson** [comment]
We tried that when we were designing the Phenomizer algorithm and it did not work well. There is no real reason to think that the various phenotypes should match all of the phenotypes of the query, although perhaps there are subprofiles that could be exploited. It is an interesting topic.


Peter Robinson

Professor of Computational Biology

The Jackson Laboratory for Genomic Medicine

10 Discovery Drive

Farmington, CT 06032

860.837.2095 t | 860.990.3130 m

peter.robinson@jax.org<mailto:peter.robinson@jax.org>

www.jax.org

Robinson lab: https://robinsongroup.github.io/

The Jackson Laboratory: Leading the search for tomorrow's cures

________________________________

**Comment by jmcmurry** [comment]
> There is no real reason to think that the various phenotypes should match all of the phenotypes of the query

I agree; this is why I was just talking about those above a certain threshold, otherwise profiles with lots of phenotypes would be unfairly penalized. Right now, we have that profiles with lots of phenotypes are potentially already penalized due to the number of those unmatched. However, if they're not completely unmatched, but rather fuzzy matched, it could change the calculus rather a lot.

**Comment by cmungall** [comment]
I agree with Peter, matching all phenotypes against all doesn't work.

But I see the intuition guiding the question and this deserves an answer along the lines of having a probabilistic model that can account for a latent phenotype explaining multiple similar phenotypes. Not sure I can write it in a GH ticket though.

Also, are you totally sure the figure is an accurate depiction of what is coming back from the algorithm? I would expect GDD and DSaLD to pair.
